src/context/CategoriesProvider.jsx

```
@@ -0,0 +1,37 @@
+import axios from "axios";
+import { useState, useEffect, createContext } from "react";
+
+// ----------------------------------------------------------------------
+
+const CategoriesContext = createContext();
+
+// ----------------------------------------------------------------------
+
+const CategoriesProvider = ({ children }) => {
+  const [categories, setCategories] = useState([]);
+
+  const getCategories = async () => {
+    try {
+      const url = "https://www.thecocktaildb.com/api/json/v1/1/list.php?c=list";
+      const { data } = await axios(url);
+
+      setCategories(data.drinks);
+    } catch (error) {
+      console.log(error);
```

Use "throw new Error", instead of printing the error on the console.